Packs/QRadarXSOARIncidentConcurrency/Scripts/QRadarXSOARIncidentConcurrency/QRadarXSOARIncidentConcurrency.py

```diff
@@ -0,0 +1,135 @@
+
+    search_arg_for_demisto['notstatus'] = not_status_matched_incident
+    search_arg_for_demisto['size'] = size
+    respmatched_incidentsent = demisto.executeCommand('SearchIncidents', search_arg_for_demisto)
+    if isError(respmatched_incidentsent[0]):
+        return_error(respmatched_incidentsent)
+    else:
+        incident_from_demisto = demisto.get(respmatched_incidentsent[0], 'Contents')
+        if not incident_from_demisto:
+            error_messages = "Data don't exist for this query on XSOAR"
+            return_error(error_messages)
+        else:
+            incident_from_demisto = incident_from_demisto if isinstance(incident_from_demisto, list) else [incident_from_demisto]
+
+    matched_incidents = []
+    mismatched_incidents = []
+    k = 0
+    for data in incident_from_demisto:
+        i = 0
+        for incID in data['labels']:
+            if incID['type'] == 'id':
+                k = k + 1
+                demistoIncValuefromQradar = int(data['labels'][i]['value'])  # type transformation from Unicode to int
+                break
+            else:
+                i = i + 1
+        matched_incidents.append(demistoIncValuefromQradar)
+        demistoIncValue = int(data['id'])
+        mismatched_incidents.append(demistoIncValue)
+
+    return mismatched_incidents, matched_incidents
+
+
+def controlQRadarXSOAR(qradar_offenses, mismatched_incidents, matched_incidents):
+    qradar_id_array = qradar_offenses
+    mismatched_incidents = mismatched_incidents
+    matched_incidents = matched_incidents
+    # Find Different Incident and Close it on QRadar
+
+    difference_incident = []  # Different element between Demisto and QRadar
+    match_incidents_list = []
+    i = 0
+    k = 0
+    for demisto_element in matched_incidents:
+        is_different = 0  # 0 is no different, 1 is different
+
+        for qradar_element in qradar_id_array:
+            if demisto_element == qradar_element:
+                is_different = 1
+                break
+
+        if is_different != 1:
+            difference_incident.append(mismatched_incidents[i])
+        else:
+            match_incidents_list.append(matched_incidents[k])
+        i = i + 1
+        k = k + 1
+
+    mach_incident_count = 'Count of matching events between QRadar and SOAR : ' + str(len(match_incidents_list))
+    demisto.results(mach_incident_count)
+    mismatched_incident_count = "Number of incidents inconsistent between QRadar and XSOAR : " + str(len(difference_incident))
+    demisto.results(mismatched_incident_count)
+
+    if len(difference_incident) > 0:
+        mismatched_incident = "Incidents that are inconsistent between QRadar and XSOAR : " + str(difference_incident)
+        demisto.results(mismatched_incident)
+        return difference_incident
+    else:
+        return_messages = "No inconsistencies between QRadar and XSOAR"
+        return_results(return_messages)
+        sys.exit(1)
```

Comment:
```suggestion
```
Please don't use sys.exit().
Why are you exiting here?

Reply:
Yes, you are right, I fixed it the following code:
in **function**:
`if len(difference_incident) > 0:
        mismatched_incident = "Incidents that are inconsistent between QRadar and XSOAR : " + str(difference_incident)
        demisto.results(mismatched_incident)
        return difference_incident
    else:
        return None`
in **main**:
`if difference_incident:
        synchronizationQRadarXSOAR(difference_incident)
    else:
        return_messages = "No inconsistencies between QRadar and XSOAR"
        return_results(return_messages)`